### PR TITLE
refactor: eip1497 updated support

### PR DESCRIPTION
**Pull request by fnanni-0**
Updated archon: dynamic script proper handling, reserved rulings, etc.

### CODE REVIEW

**Review by alcercu — APPROVED** [review]
LGTM 🚀

File: README.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [![Netlify Status](https://api.netlify.com/api/v1/badges/0708f99a-542e-416d-83ce-4ae4f11e9b79/deploy-status)](https://app.netlify.com/sites/centralizedarbitrator/deploys)
 
-Supported netwoks: Mainnet, Kovan, Ropsten
+Supported netwoks: Mainnet, Kovan, Ropsten, Rinkeby.
```
Comment (Tanz0rz) [inline README.md:3]: Do you also want to list Goerli here?

Comment (fnanni-0) [reply README.md:3]: Thanks! Yes, it's actually supported already. I just fixed the readme.

File: src/ethereum/web3.js [hunk 2]
```diff
@@ -9,14 +9,14 @@ window.addEventListener('load', async () => {
...
-      await window.ethereum.enable()
+      await window.ethereum.request({ method: 'eth_requestAccounts' })
       // Acccounts now exposed
     } catch (_) {
       // User denied account access...
     }
   }
   // Legacy dapp browsers...
-  else if (window.web3) window.web3 = new Web3(web3.currentProvider)
+  else if (window.web3) window.web3 = new Web3(window.ethereum)
```
Comment (n1c01a5) [inline src/ethereum/web3.js:19]: `window.ethereum` is `undefined`. It's `new Web3(window.web3.currentProvider)`.

File: src/ethereum/web3.js [hunk 3]
```diff
@@ -25,17 +25,34 @@ window.addEventListener('load', async () => {
...
 } else {
-  console.warn(
-    "Couldn't find the web3 object of the window, falling back to Infura."
-  )
-  Web3.providers.HttpProvider.prototype.sendAsync =
-    Web3.providers.HttpProvider.prototype.send
-  const provider = new Web3.providers.HttpProvider(INFURA_ENDPOINT)
-  web3 = new Web3(provider)
+  // Fallback provider.
+  web3 = new Web3(getReadOnlyRpcUrl);
```
Comment (n1c01a5) [inline src/ethereum/web3.js:32]: Don't you have to call the function here ?

File: src/ethereum/web3.js [hunk 4]
```diff
@@ -25,17 +25,34 @@ window.addEventListener('load', async () => {
     )
 })
 
-if (typeof window !== 'undefined' && typeof window.web3 !== 'undefined') {
-  console.log('Using the web3 object of the window...')
-  web3 = new Web3(window.web3.currentProvider)
+if (typeof window !== "undefined" && typeof window.web3 !== "undefined") {
```
Comment (n1c01a5) [inline src/ethereum/web3.js:28]: What happens if `window.ethereum` is undefined ? . What is the interest of instantiating it with `undefined` ? Wouldn't it be better to display a message to install Metamask?

**Pull request merged**